### PR TITLE
[map] update geotiff flag so that it's in the url args instead of the url params

### PR DESCRIPTION
**Pull request by chejennifer**
- this allows flask to read the flag and only load the leaflet css if leaflet flag is set

### CODE REVIEW

File: static/js/tools/map/app.tsx [hunk 1]
```diff
@@ -114,9 +115,16 @@ function updateHash(context: ContextType): void {
...
+  if (context.display.value.allowLeaflet) {
+    args += `?${ALLOW_LEAFLET_URL_ARG}=1`;
+  }
   const newHash = encodeURIComponent(hash);
   const currentHash = location.hash.replace("#", "");
-  if (newHash && newHash !== currentHash) {
-    history.pushState({}, "", `${MAP_REDIRECT_PREFIX}#${newHash}`);
+  const currentArgs = location.search;
+  if (newHash && (newHash !== currentHash || args !== currentArgs)) {
+    history.pushState({}, "", `${MAP_REDIRECT_PREFIX}${args}#${newHash}`);
```
Comment (beets) [inline static/js/tools/map/app.tsx:128]: We should forward along all args (not just leaflet), and then append the hash. Query args are a good place for persistent effects (e.g. allow various features). This works for now, and we can update that in a TODO.

Perhaps also rename `MAP_REDIRECT_PREFIX` to `MAP_URL_PATH` so we don't accidentally add other args / path to the const.

File: static/js/tools/map/util.ts [hunk 2]
```diff
@@ -344,7 +342,11 @@ export function getRedirectLink(
     parentPlaces: [],
     selectedPlace,
   });
-  return `${MAP_REDIRECT_PREFIX}#${encodeURIComponent(hash)}`;
+  let args = "";
+  if (displayOptions.allowLeaflet) {
+    args += `?${ALLOW_LEAFLET_URL_ARG}=1`;
+  }
+  return `${MAP_REDIRECT_PREFIX}${args}#${encodeURIComponent(hash)}`;
```
Comment (beets) [inline static/js/tools/map/util.ts:349]: please add a manual comment equivalent to IFTT to update both files when either is edited

**Review by beets — APPROVED** [review]

File: static/js/tools/map/app.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/tools/map/app.tsx]: added todo & renamed the constant

File: static/js/tools/map/util.ts (hunk 2 above)
Comment (chejennifer) [reply static/js/tools/map/util.ts]: done

**Pull request merged**